pkg/ccl/backupccl/restore_job.go

```diff
@@ -1072,6 +1072,49 @@ func createImportingDescriptors(
 				}
 			}
 
+			// We could be restoring a TTL table, in which case we need to allocate
+			// a new schedule to it.
+			if details.DescriptorCoverage != tree.AllDescriptors {
+				for _, table := range mutableTables {
+					if table.HasRowLevelTTL() {
+						shouldCreateScheduledJob := false
+						existingJob, err := jobs.LoadScheduledJob(
+							ctx,
+							sql.JobSchedulerEnv(p.ExecCfg()),
+							table.RowLevelTTL.ScheduleID,
+							p.ExecCfg().InternalExecutor,
+							txn,
+						)
+						if err != nil {
+							if !jobs.HasScheduledJobNotFoundError(err) {
+								return errors.Wrapf(err, "unknown error fetching existing job for row level TTL in schema changer")
+							}
+							shouldCreateScheduledJob = true
+						} else {
+							var args catpb.ScheduledRowLevelTTLArgs
+							if err := pbtypes.UnmarshalAny(existingJob.ExecutionArgs().Args, &args); err != nil || args.TableID != table.GetID() {
+								shouldCreateScheduledJob = true
+							}
+						}
+
+						if shouldCreateScheduledJob {
+							j, err := sql.CreateRowLevelTTLScheduledJob(
```

Comment:
Do we need to clean these up in OnFailOrCancel? I see that these jobs are automatically dropped in the schema changer when we process a dropped table descriptor, but our dropDescriptor method seems to do a lot of the heavy lifting itself so it isn't clear to me that the logic in the schema changer will be in play here.  

If this is automatically handled and I've just misread things, perhaps we could add a test to ensure they are cleaned up?

Reply:
`TestRestoreFailCleansUpTypeBackReferences` or `TestRestoreFailDatabaseCleanup` look like they might be useful to start from.

Reply:
done!